text/0000-assoc-default-groups.md

```diff
@@ -0,0 +1,957 @@
+- Feature Name: `assoc_default_groups`
+- Start Date: 2018-08-23
+- RFC PR: _
+- Rust Issue: _
+
+# Summary
+[summary]: #summary
+
+[RFC 192]: https://github.com/rust-lang/rfcs/blob/master/text/0195-associated-items.md#defaults
+
+1. [Resolve][changes] the design of associated type defaults,
+   first introduced in [RFC 192],
+   such that provided methods and other items may not assume type defaults.
+   This applies equally to `default` with respect to specialization.
+
+2. [Introduce][default_groups] the concept of `default { .. }` groups
+   in traits and their implementations which may be used to introduce
+   atomic units of specialization
+   (if anything in the group is specialized, everything must be).
+
+# Motivation
+[motivation]: #motivation
+
+## For associated type defaults
+
+As discussed in the [background] and mentioned in the [summary],
+associated type defaults were introduced in [RFC 192].
+These defaults are valuable for a few reasons:
+
+1. You can already provide defaults for `const`s and `fn`s.
+   Allowing `type`s to have defaults adds consistency and uniformity
+   to the language, thereby reducing surprises for users.
+
+2. Associated `type` defaults in `trait`s simplify the grammar,
+   allowing the grammar of `trait`s them to be more in line with
+   the grammar of `impl`s. In addition, this brings `trait`s more in line
+   with `type` aliases.
+
+The following points were also noted in [RFC 192], but we expand upon them here:
+
+3. Most notably, type defaults allow you to provide more ergonomic APIs.
+
+   [proptest]: https://altsysrq.github.io/rustdoc/proptest/latest/proptest/arbitrary/trait.Arbitrary.html
+
+   For example, we may provide an API (due to [proptest]):
```

Comment:
It might read better to write this line as:

`For example, we could change [proptest]'s API to be:`